### ISSUE TITLE
Porting to other platforms?

### ISSUE THREAD

**Issue by TimLeijten** [chocolate-doom/chocolate-doom#1152]
Hi,

So I want to port chocolate doom to some other development board that don't run an os, but support all C and C++17 functionality. I'm wondering, what do I need to change in order to make it run? There is NO SDL support, audio support, input support whatso-ever. So I need to implement that myself(which is fine), but where would I find that stuff? Is there like a list of files containing non-C stuff(files that use sdl and such)? That would make my life much easier.

So basically what I'm asking is, what libraries like SDL does chocolate doom use which I need to replace, and where would those implementations be?(Implementations being the parts of choco doom that use those libraries)

Thanks!

**Comment by fragglet** [comment]
Pretty much the only files you should need to change are the system-specific files - those named `i_*`.

However, you might be better off seeing if you can port LibSDL to the platform instead, because then you can easily port other programs besides just Doom. SDL has dummy backends which should allow you to compile it even if there's nothing system-specific you can use. Then you can go about writing the individual modules for video, input, etc.

**Comment by TimLeijten** [comment]
> Pretty much the only files you should need to change are the system-specific files - those named `i_*`.
> 
> However, you might be better off seeing if you can port LibSDL to the platform instead, because then you can easily port other programs besides just Doom. SDL has dummy backends which should allow you to compile it even if there's nothing system-specific you can use. Then you can go about writing the individual modules for video, input, etc.

Thanks! Proting sdl might be an idea later, but currently I just want to port doom. Thanks though!

**Comment by fragglet** [comment]
You'll probably find it's easier to port SDL than it is to replace all the `i_*` files anyway. But it's up to you how you want to proceed. Good luck!
